judge/views/api/api_v2.py

```diff
@@ -1,122 +1,593 @@
+            'format': {
+                'name': contest.format_name,
+                'config': contest.format_config,
+            },
+            'problems': [
+                {
+                    'points': int(problem.points),
+                    'partial': problem.partial,
+                    'is_pretested': problem.is_pretested and contest.run_pretests_only,
+                    'max_submissions': problem.max_submissions or None,
+                    'label': contest.get_label_for_problem(index),
+                    'name': problem.problem.name,
+                    'code': problem.problem.code,
+                } for index, problem in enumerate(problems)
+            ] if can_see_problems else [],
+            'rankings': [
+                {
+                    'user': participation.username,
+                    'score': participation.score,
+                    'cumulative_time': participation.cumtime,
+                    'tiebreaker': participation.tiebreaker,
+                    'old_rating': participation.old_rating,
+                    'new_rating': participation.new_rating,
+                    'is_disqualified': participation.is_disqualified,
+                    'solutions': contest.format.get_problem_breakdown(participation, problems),
+                } for participation in participations
+            ] if can_see_rankings else [],
+        }
+
+
+class APIContestParticipationList(APIListView):
+    model = ContestParticipation
+    basic_filters = (
+        ('contest', 'contest__key'),
+        ('user', 'user__user__username'),
+        ('is_disqualified', 'is_disqualified'),
+        ('virtual_participation_number', 'virtual'),
+    )
+
+    def get_unfiltered_queryset(self):
+        visible_contests = Contest.get_visible_contests(self.request.user)
+
+        # Check which contest scoreboards the user can actually see.
+        # "Contest.get_visible_contests" only gets which contests the user can *see*.
+        # Conditions for participation scoreboard access:
+        #   1. Contest has ended
+        #   2. User is the organizer of the contest
+        #   3. User is specified to be able to "view contest scoreboard"
+        if not self.request.user.has_perm('judge.see_private_contest'):
+            q = Q(end_time__lt=self._now)
+            if self.request.user.is_authenticated:
+                if self.request.user.has_perm('judge.edit_own_contest'):
+                    q |= Q(organizers=self.request.profile)
+                q |= Q(view_contest_scoreboard=self.request.profile)
+            visible_contests = visible_contests.filter(q)
+
+        return (
+            ContestParticipation.objects
+            .filter(virtual__gte=0, contest__in=visible_contests)
+            .annotate(contest_key=F('contest__key'), username=F('user__user__username'))
+            .order_by('id')
+            .only('score', 'cumtime', 'tiebreaker', 'is_disqualified', 'virtual')
```

Comment:
Might be more efficient to just use `values` and make `get_object_data` identity.